### PR TITLE
Refactor TCP connection logic

### PR DESCRIPTION
**Pull request by rukai**
prereq to https://github.com/shotover/shotover-proxy/pull/899

Every connection needs this same basic error handling so we should move it into a helper to give us consistent error messages and to ensure we never forget to implement the timeout.
It will also allow us to call this logic from within the tls connector in 899.

### CODE REVIEW

File: shotover-proxy/src/tcp.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+use anyhow::{anyhow, Result};
+use std::time::Duration;
+use tokio::{
```
Comment (conorbros) [inline shotover-proxy/src/tcp.rs:3]: Could we rename `tls.rs` to `transport.rs` and put all the TCP/TLS stuff in there?

Comment (rukai) [reply shotover-proxy/src/tcp.rs:3]: IMO tls has enough going on to be its own module.
Would you want to do transport/tls and trasnsport/tcp ?

Comment (conorbros) [reply shotover-proxy/src/tcp.rs:3]: No not that fussed. Just a thought. Can merge as is.

File: shotover-proxy/src/tcp.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    net::{TcpStream, ToSocketAddrs},
+    time::timeout,
+};
+
+pub async fn tcp_stream<A: ToSocketAddrs + std::fmt::Debug>(destination: A) -> Result<TcpStream> {
+    timeout(Duration::from_secs(3), TcpStream::connect(&destination))
```
Comment (benbromhead) [inline shotover-proxy/src/tcp.rs:9]: Given this is a refactor, we should probably address making this timeout configurable with a 3 sec default. I can't imagine why we would need to longer, but shorter might be appropriate for some cases.

Comment (rukai) [reply shotover-proxy/src/tcp.rs:9]: That is quite a large change so I've done it in a separate PR with this PR as a prereq: https://github.com/shotover/shotover-proxy/pull/901

**Review by benbromhead — APPROVED** [review]
LGTM

**Pull request merged**